src/core/service.c

```diff
@@ -4760,7 +4760,7 @@ int service_set_socket_fd(
                                             "Failed to add After=/TriggeredBy= dependencies on socket unit: %m");
 
         s->socket_fd = fd;
-        s->socket_peer = socket_peer_ref(peer);
+        s->socket_peer = peer;
```

Comment:
hmm can you add a comment about this to the parameter declaration above? if objects are donated this deserves a comment on the parameter.

i.e. just add `/* object donated to us on success */`

Reply:
I guess it's quicker to just prep the fix myself:

https://github.com/systemd/systemd/pull/32174